pyvrp/cpp/TimeWindowSegment.h

```diff
@@ -105,13 +105,10 @@ class TimeWindowSegment
                              Duration releaseTime);
 };
 
-TimeWindowSegment TimeWindowSegment::merge(
-    [[maybe_unused]] Matrix<Duration> const &durationMatrix,
-    [[maybe_unused]] TimeWindowSegment const &other) const
+TimeWindowSegment
+TimeWindowSegment::merge(Matrix<Duration> const &durationMatrix,
+                         TimeWindowSegment const &other) const
 {
```

Comment:
Method's private and only called from the public method with the same name. That method already has these flags, so there's no need to have them here, too.